### ISSUE TITLE
[explicit-function-return-type] allowTypedFunctionExpressions does not work for parameters

### ISSUE THREAD

**Issue by davidje13** [typescript-eslint/typescript-eslint#679]
**Repro**

```json
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "sourceType": "module"
  },
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/explicit-function-return-type": ["error", {
      "allowTypedFunctionExpressions": true,
    }]
  }
}
```

```typescript
class Accumulator {
  private count: number = 0;

  public accumulate(fn: () => number): void {
    this.count += fn();
  }
}

new Accumulator().accumulate(() => 1);
```

**Expected Result**

No warnings; type of `() => 1` is inferred from `accumulate` method signature, due to use of `allowTypedFunctionExpressions`

**Actual Result**

> ```
> error: Missing return type on function (@typescript-eslint/explicit-function-return-type) at foobar/Accumulator.ts:9:30:
>    7 | }
>    8 | 
> >  9 | new Accumulator().accumulate(() => 1);
>      |                              ^
>   10 | 
> ```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.11.0` |
| `@typescript-eslint/parser`        | `1.11.0` |
| `TypeScript`                       | `3.5.2` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.4.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
Apparently this wasn't ever added to the implementation. I've added handling for it in #680
